apps/src/craft/craft.js

```diff
@@ -238,86 +239,92 @@ Craft.init = function (config) {
       break;
   }
 
-  studioApp.init($.extend({}, config, {
-    forceInsertTopBlock: 'when_run',
-    html: require('../templates/page.html.ejs')({
-      assetUrl: studioApp.assetUrl,
-      data: {
-        localeDirection: studioApp.localeDirection(),
-        visualization: require('./visualization.html.ejs')(),
-        controls: require('./controls.html.ejs')({
-          assetUrl: studioApp.assetUrl,
-          shareable: config.level.shareable
-        }),
-        editCode: config.level.editCode,
-        blockCounterClass: 'block-counter-default',
-        readonlyWorkspace: config.readonlyWorkspace
-      }
-    }),
-    appStrings: {
-      generatedCodeDescription: craftMsg.generatedCodeDescription(),
-    },
-    loadAudio: function () {
+  React.render(React.createElement(AppView, {
+    renderCodeApp: function () {
+      return require('../templates/page.html.ejs')({
```

Comment:
I don't like requires in the middle of the file like this. I realize this isnt new, so your discretion as to whether you want to change or not.